fire/SFMainMod.F90

```diff
@@ -688,10 +689,14 @@ subroutine  area_burnt_intensity ( currentSite )
     
     ! Equation 7 from Venevsky et al GCB 2002 (modification of equation 8 in Thonicke et al. 2010) 
     ! FDI 0.1 = low, 0.3 moderate, 0.75 high, and 1 = extreme ignition potential for alpha 0.000337
-    currentSite%FDI  = 1.0_r8 - exp(-SF_val_fdi_alpha*currentSite%acc_NI)
+    if (use_fates_spitfire == 3) then
+       currentSite%FDI = 1.0_r8  ! READING "SUCCESSFUL IGNITION" DATA
+    else  ! USING LIGHTNING DATA
+       currentSite%FDI  = 1.0_r8 - exp(-SF_val_fdi_alpha*currentSite%acc_NI)
+    end if
     
     !NF = number of lighting strikes per day per km2 scaled by cloud to ground strikes
-    currentSite%NF = ED_val_nignitions * years_per_day * cg_strikes
```

Comment:
There's some discussion on what this should look like on the CTSM side (should the no-data option fill the 3D array for lightning24? This is really the crux of that issue here. You can keep this as one line of code if you make the constant option fill the array, which is easier to read. But, it's also more expensive computationally for multiplying by a constant value. So I think it's OK to have an if statement here to cover the old global constant versus the new way with temporally and spatially varying data.

I was going to suggest that you make the constant used -- comparable to the lightning data, but it's already the same only multiplied by a single constant, so I think it's OK to leave as is. You could easily multiply the constant by 365 in order to see how it compares to the lightning data. Although, I guess the native lightning data is hourly rather than daily, so there's another factor of 24 there.

But, there's also value in having an if statement here so that you can show the previous method is identical to what it was before (when you remove white space differences). It is bad to have tons of logical statements scattered throughout the code, but sometimes it's necessary...

@jkshuman can you comment on this point?

Reply:
Thanks @ekluzek. Retaining an if statement for the case where there is no lightning dataset and we use the constant value as written in SPITFIRE would be useful. It is not necessary to have an array of these values, so the if statement takes care of that.

Reply:
Regarding the need for if statements, I presented two options [here](https://github.com/ESCOMP/CTSM/pull/991#discussion_r417634740).

Reply:
I have reconsidered my previous decision to put the if-statement in clmfates_interface and have put it here instead. This way I don't need to use ED_val_nignitions in clmfates_interface, so as to remain consistent with the goal of keeping the two models as distinct as possible.